tests/rados/test_osd_inconsistency_pg.py

```diff
@@ -0,0 +1,110 @@
+        log.info(f"The object-{oname} is created in the pool-{pool_name}")
+        osd_map_output = rados_obj.get_osd_map(pool=pool_name, obj=oname)
+        primary_osd = osd_map_output["acting_primary"]
+        log.info(f"The object stored in the primary osd number-{primary_osd}")
+        pg_id = osd_map_output["pgid"]
+        log.info(f"The object {oname} is created in the pg-{pg_id}")
+        # Adding the key value to object
+        log.info("Adding a key to the object")
+        cmd_setkey = f"rados -p {pool_name} setomapval {oname} key0 value0"
+        client_node.exec_command(cmd=cmd_setkey, sudo=True)
+        # stoping the OSD
+        if not rados_obj.change_osd_state(action="stop", target=primary_osd):
+            log.error(f"Unable to stop the OSD : {primary_osd}")
+            raise Exception("Execution error")
+        acting_osd_node = rados_obj.fetch_host_node(
+            daemon_type="osd", daemon_id=primary_osd
+        )
+        cmd_base = f"cephadm shell --name osd.{primary_osd} --"
+        log.info(f"Corrupting the object-{oname} key.")
+        cmd_get_snaObj = (
+            f"{cmd_base} ceph-objectstore-tool --data-path /var/lib/ceph/osd/ceph-"
+            f"{primary_osd} --pgid {pg_id} {oname} set-omap key_corrupt1 /etc/hosts"
+        )
+        acting_osd_node.exec_command(sudo=True, cmd=cmd_get_snaObj)
+        if not rados_obj.change_osd_state(action="start", target=primary_osd):
+            log.error(f"Unable to start the OSD : {primary_osd}")
+            raise Exception("Execution error")
+            # Wait until pg active+clean state
+        while True:
+            pg_report = rados_obj.check_pg_state(pgid=pg_id)
+            if "active+clean" in pg_report:
+                break
+            log.info(f"The pg-{pg_id} is not in active+clean state")
+            time.sleep(10)
+        log.info(f"The pg-{pg_id} is in active+clean state")
+        # Perform deep-scrub
+        log.info(f"Performing the deep-scrub on the pg-{pg_id}")
+        rados_obj.run_deep_scrub(pgid=pg_id)
+        # sleeping for 10 seconds, waiting for scrubbing to be over
+        time.sleep(10)
+        # Checking the cluster health status
+        health_detail, _ = cephadm.shell(args=["ceph health detail"])
+        assert "inconsistent*" not in health_detail
+        log.info("pg inconsistent generated in the cluster")
+        log.info(f"Health warning: \n {health_detail}")
+        # Checking the inconsistent PG with the pool name
+        cmd_inconsist_pg = f"rados list-inconsistent-pg  {pool_name}"
+        inconsistent_pg_list = rados_obj.run_ceph_command(cmd_inconsist_pg)
+        if any(pg_id in search for search in inconsistent_pg_list):
+            log.info("Inconsistent PG list exist ")
+        else:
+            log.error("Inconsistent PG is not generated")
+            return 1
+        # checking the  inconsistent objects
+        cmd_inconsist_object = f"rados  list-inconsistent-obj  {pg_id}"
```

Comment:
Please write a method for list-inconsistent-obj so that we can reuse the same in other tests for inconsistency.

Reply:
Method added.